cli/flox/doc/manifest.toml.md

```diff
@@ -0,0 +1,310 @@
+but may also be specified either at install-time via the `-i` option
+or interactively via `flox edit`.
+
+The value in the key-value pair is called a "package descriptor".
+A package is specified by a number of available options which are separate
+from the install ID,
+so you are free to change them independently of one another.
+This allows you to change package details while keeping a stable install ID,
+for example upgrading from `gcc.pkg-path = "gcc12"` to
+`gcc.pkg-path = "gcc13"`.
+
+The descriptor options allow you to specify in detail the package to install.
+The full list of descriptor options are shown below:
+```
+Descriptor ::= {
+  name               = null | <STRING>
+, optional           = null | <BOOL>
+, pkg-group          = null | <STRING>
+, version            = null | <STRING>
+, semver             = null | <STRING>
+, systems            = null | [<STRING>, ...]
+, pkg-path           = null | <STRING> | [<STRING>, ...]
+, abs-path           = null | <STRING> | [<STRING>, ...]
+, priority           = null | <INT>
+}
+```
+
+None of these options are required,
+and leaving them unset instructs Flox to simply find the best match for the
+package name and latest version given the install ID.
+In order to leave all options unset,
+simply use an empty package descriptor:
+
+```
+ripgrep = {}
+```
+
+By specifying some of these options you create a set of requirements that the
+installed program must satisfy,
+otherwise installation will fail.
+The most common option will likely be the `semver` option,
+which allows you to specify a semantic version range.
+
+Each option is described below:
+
+`name`
+:   Matches either the last attribute of the `pkg-path` or the package metadata
+    fields `name` or `pname` as set by the catalog.
+    This option is mutually exclusive with the `pkg-path` and `abs-path`
+    options.
+    You shouldn't need to use this option and should instead prefer the
+    `pkg-path` option.
+
+`optional`
+:   Marks this package as only an optional requirement for the environment.
+    By default an environment will fail to build if a specified package can't
+    be found in the catalog.
+    However, some packages are platform specific and will never be found in the
+    catalog for some systems.
+    Thus, the way you mark a package as platform specific is by setting
+    `opitonal = true` and using the `systems` option to list the systems on
+    which the package is required.
+
+`pkg-group`
+:   Marks a package as belonging to a group of packages.
+    Some packages only work together correctly when they all come from the
+    same revision of the catalog.
+    One example is C/C++ projects that depend on specific versions of header
+    files.
+    Packages are marked as belonging to a group simply by setting this option
+    to the name of the group.
+
+    Packages that are marked as belonging to the same group will always be
+    installed from the same revision of the catalog.
```

Comment:
nonblocking: can we describe this in a less implementation specific way? Are guaranteed to be compatible or something? If we're going to use the word revision, do we need to define it?

Reply:
I think this is fine for the amount of knowledge we have about what the catalog even looks like. We can always change the man page once the catalog is no longer 1:1 with nixpkgs.

I hesitate to say "guaranteed to be compatible" because that makes it sound like other packages aren't guaranteed to work together.

Reply:
would probably be better to talk about the general idea: "...the same group will be optimized for compatibility and dependency sharing."

Reply:
I'm not sure that's any more clear than what I currently have.

Reply:
I think something like "adding packages to a group ensures all packages share the same libraries and dependencies, which keeps the environment size down" explains why you would want something in the same group, whereas "installed from the same revision of the catalog" is kind of meaningless without more in-depth understanding of what that means